[Fix #10202] Add new `Lint/Ruby2Keywords` cop

dvandersluis opened the pull request:
Looks for instances of `ruby2_keywords` for methods that do not need it. 

`ruby2_keywords` is only needed for methods that have a splat (`*args`) and no keyword args (regular, optional or keyword splat).

Can handle three forms:

```ruby
ruby2_keywords def foo(arg)
end
```

```ruby
def foo(arg)
end
ruby2_keywords :foo
```

```ruby
define_method(:foo) { |arg| }
ruby2_keywords :foo
```

Fixes #10202.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

koic commented on lib/rubocop/cop/lint/ruby2_keywords.rb:
```diff
@@ -0,0 +1,111 @@
+      # (`*args`) but do not explicit keyword arguments (`k:` or `k: true`) or
+      # a keyword splat (`**kwargs`).
+      #
+      # @example `ruby2_keywords` given a `def`
```
`@example` directive is used to represent `EnforcedStyle` (and similar style option). Can you use source comment for explanation to examples? :-)

koic commented on lib/rubocop/cop/lint/ruby2_keywords.rb:
```diff
@@ -0,0 +1,111 @@
+      #   ruby2_keywords :foo
+      #
+      #   # good
+      #   def foo(*args)
+      #   end
+      #   ruby2_keywords :foo
```
A good case compatible with a bad case would be to remove `ruby2_keywords`.

```suggestion
      #   def foo
      #   end
```

koic replied:
And I think it will be easier to understand whether this case is represented at the top of example.

koic commented on lib/rubocop/cop/lint/ruby2_keywords.rb:
```diff
@@ -0,0 +1,111 @@
+      #   ruby2_keywords :foo
+      #
+      #   # good
+      #   define_method(:foo) { |*args| }
+      #   ruby2_keywords :foo
```
```suggestion
      #   define_method(:foo) { |arg| }
```

koic commented:
I think the cop name `UselessRuby2Keywords` cop is better.

dvandersluis commented:
@koic I renamed the cop and changed the examples. I still think it's not a bad thing to have separate @example blocks for different use cases (I understand that we usually do this for configuration examples, but still), but I rewrote it to be more like you were suggesting.
